gensim/models/wrappers/varembed.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (C) 2017 Anmol Gulati <anmol01gulati@gmail.com>
+# Copyright (C) 2017 Radim Rehurek <radimrehurek@seznam.cz>
+
+"""
+Python wrapper around word representation learning from Varembed models, a library for efficient learning of word representations and sentence classification [1].
+
+This module allows ability to obtain word vectors for out-of-vocabulary words, for the Varembed model[2].
+
+The wrapped model can NOT be updated with new documents for online training -- use gensim's `Word2Vec` for that.
+
+.. [1] https://github.com/rguthrie3/MorphologicalPriorsForWordEmbeddings
+
+.. [2] http://arxiv.org/pdf/1608.01056.pdf
+"""
+
+import logging
+
+import numpy as np
+
+from gensim.models.keyedvectors import KeyedVectors
+from gensim.models.word2vec import Word2Vec
+
+# utility fnc for pickling, common scipy operations etc
+from gensim import utils
+from gensim.models.word2vec import Vocab
+
+logger = logging.getLogger(__name__)
+
+
+class VarEmbed(Word2Vec):
+    """
+    Class for word vectors using Varembed models. Contains methods to load a varembed model and implements
+    functionality like `most_similar`, `similarity` by extracting vectors into numpy matrix.
+    Refer to [Varembed]https://github.com/rguthrie3/MorphologicalPriorsForWordEmbeddings for
+    implementation of Varembed models.
+    """
+
+    def __init__(self):
+        self.wv = KeyedVectors()
+        self.vector_size = 0
+        self.vocab_size = 0
+
+    @classmethod
+    def load_varembed_format(cls, vectors, morfessor_model=None, use_morphemes=False):
+        """
+        Load the input-hidden weight matrix from the fast text output files.
+
+        Note that due to limitations in the FastText API, you cannot continue training
+        with a model loaded this way, though you can query for word similarity etc.
+
+        'vectors' is the pickle file containing the word vectors.
+        'morfessor_model' is the path to the trained morfessor model.
+        'use_morphemes' False(default) use of morpheme embeddings in output.
+        """
+        result = cls()
+        if vectors is None:
+            raise Exception(
+                "Please provide vectors binary to load varembed model")
+        D = utils.unpickle(vectors)
+        word_to_ix = D['word_to_ix']
+        morpho_to_ix = D['morpho_to_ix']
+        word_embeddings = D['word_embeddings']
+        morpho_embeddings = D['morpheme_embeddings']
+        result.load_word_embeddings(word_embeddings, word_to_ix)
+        if use_morphemes:
+            try:
+                import morfessor
+                morfessor_model = morfessor.MorfessorIO().read_binary_model_file(morfessor_model)
```

Comment:
Yes, Sounds good. I've put up an issue  https://github.com/rguthrie3/MorphologicalPriorsForWordEmbeddings/issues/3 to notify them about the new release on morfessor as well.